Atomic/syscontainers.py

```diff
@@ -1496,37 +1522,24 @@ def filter_func(*args):
 
             modifier = OSTree.RepoCommitModifier.new(0, filter_func, None)
 
-            metav = GLib.Variant("a{sv}", {'docker.layer': GLib.Variant('s', layer)})
+            checkout = self._get_system_checkout_path()
+            destdir = checkout if os.path.exists(checkout) else None
 
-            imported = False
             try:
-                repo.write_archive_to_mtree(Gio.File.new_for_path(tar), mtree, modifier, True)
+                temp_dir = tempfile.mkdtemp(prefix=".", dir=destdir)
+                # NOTE: tarfile has an issue with utf8. This works around the problem
+                # by using the systems tar command.
+                # Ref: https://bugzilla.redhat.com/show_bug.cgi?id=1194473
+                subprocess.check_call(['tar', '-xf', tar, '-C', temp_dir])
+                repo.write_directory_to_mtree(Gio.File.new_for_path(temp_dir), mtree, modifier)
                 root = repo.write_mtree(mtree)[1]
+
+                metav = GLib.Variant("a{sv}", {'docker.layer': GLib.Variant('s', layer),
+                                               'docker.size': get_directory_size(temp_dir)})
```

Comment:
we initially had `docker.layer` to refer to the Docker image layer, that now it is know as OCI layer.

I have used `docker.size` here since it is the same name used in the Skopeo implementation, so hopefully switching to Skopeo can be done transparently